lfs/config.go

```diff
@@ -409,16 +415,26 @@ func (c *Configuration) loadGitConfig() bool {
 	c.gitConfig = make(map[string]string)
 	c.extensions = make(map[string]Extension)
 	uniqRemotes := make(map[string]bool)
+	uniqInsteadOfs := make(map[string]string)
+	c.insteadOfs = make(map[string]string)
 
 	configFiles := []string{
 		filepath.Join(LocalWorkingDir, ".lfsconfig"),
 
 		// TODO: remove .gitconfig support for Git LFS v2.0 https://github.com/github/git-lfs/issues/839
 		filepath.Join(LocalWorkingDir, ".gitconfig"),
 	}
+
+	listOutput, err := git.Config.ListRegexp(`^url.*`)
+	if err != nil {
+		panic(fmt.Errorf("Error listing git config with regexp `^url.*`: %s", err))
+	}
+	c.readGitInsteadOfConfig(listOutput, uniqInsteadOfs)
+	c.insteadOfs = uniqInsteadOfs
```

Comment:
Does this need to call `git config` again? I wonder if it'd be better to just filter the values from the `git.Config.List()` call below? 

We try to limit the number of process execs where reasonable, especially in a spot like this that's called on every `git lfs` command. With default settings, doing an initial clone on a repository with lots of LFS objects will call `git lfs smudge` over and over again. This is especially bad on Windows, which is slower at execing a new process than linux. 

It's probably worth writing a quick benchmark so we can compare different approaches on the various platforms.